### PR TITLE
feature/TECH 532 implement manual build selection

### PR DESCRIPTION
**Pull request by Jorg88**
- [TECH-532] Add support for manual project selection to the cli
- [TECH-532] Add manual project selection to the jenkinsfile

[TECH-532]: https://vandebron.atlassian.net/browse/TECH-532?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TECH-532]: https://vandebron.atlassian.net/browse/TECH-532?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

----
### 📕 [TECH-532](https://vandebron.atlassian.net/browse/TECH-532) Figure out a manual build approach jorgpost@vandebron.nl 
Valid issues to take care of:

* changes to libraries resulting in no file changes → security issue , need to rebuild and deploy
* building only a subset
* keeping all your projects up to date
* change to common pipeline code
* building other projects from a tag

🏗️ Build [8](https://jenkins.k8s-serv-backend.vdbinfra.nl/job/MPyL%20Pipeline%20-%20Test/job/PR-301/8/display/redirect) ✅ Successful, started by _Jorg Post_  
🚀 *[cloudfront-service](https://cloudfront-service-301.test.nl/swagger/index.html)*, *example-dagster-user-code*, *job*, *kong-sync*, *[nodeservice](https://nodeservice-301.test.nl/swagger/index.html)*, *ocpp-first*, *ocpp-second*, *[sbtservice](https://sbtservice-301.test.nl/swagger/index.html)*, *sparkJob*

### CODE REVIEW

**Review by koeves — APPROVED** [review]
Looks nice and simple!

File: src/mpyl/build.py [hunk 1]
```diff
@@ -205,14 +206,20 @@ def find_build_set(
             project_paths,
         )
     )
+    if selected_projects:
+        projects_list = selected_projects.split(",")
```
Comment (koeves) [inline src/mpyl/build.py:210]: Don't forget the strip any whitespace as well

File: src/mpyl/cli/build.py [hunk 2]
```diff
@@ -149,9 +149,16 @@ def invoke(self, ctx):
...
+    "--projects",
+    "--p",
+    type=str,
+    required=False,
+    help="Comma separated list of the projects to build",
+)
 @click.pass_obj
 def run(
-    obj: CliContext, ci, all_, tag, stage, sequential
+    obj: CliContext, ci, all_, tag, stage, sequential, projects
```
Comment (koeves) [inline src/mpyl/cli/build.py:161]: By the way I would consider using the `multiple` option in click: https://click.palletsprojects.com/en/7.x/options/#multiple-options

So we don't need to parse the list ourselves

File: src/mpyl/build.py (hunk 1 above)
Comment (Jorg88) [reply src/mpyl/build.py:210]: But you can't pass whitespaces via the cli?

File: src/mpyl/cli/build.py (hunk 2 above)
Comment (Jorg88) [reply src/mpyl/cli/build.py:161]: I personally prefer `-p job,ocpp,kong-sync` from `-p job -p ocpp -p kong-sync`, and allowing both actually makes more of a mess of the parse logic I would say?

Comment (koeves) [reply src/mpyl/cli/build.py:161]: I didn't mean both. So if you prefer the single one it's also fine

Comment (koeves) [reply src/mpyl/cli/build.py:161]: does it work with `-p 'job, ocpp,kong-sync'` as well?

File: src/mpyl/build.py (hunk 1 above)
Comment (koeves) [reply src/mpyl/build.py:210]: does it work with -p 'job, ocpp,kong-sync' as well?

**Pull request merged**